src/main/java/org/apache/sling/testing/clients/SlingClient.java

```diff
@@ -580,6 +588,34 @@ public String getUUId(JsonNode jsonNode) throws ClientException {
         return uuidNode.getValueAsText();
     }
 
+    @Override
+    public String getUser() {
+        // get the username from the sudo cookie or default from client config
+        return getCookieStore().getCookies().stream().filter(c -> c.getName().equals(getSudoCookieName())).findFirst()
+                .map(c -> c.getValue().replace("\"", "")).orElse(super.getUser());
+    }
+
+    /**
+     * Impersonate user with the given <code>userId</code>
+     * <p>
+     * By impersonating a user SlingClient can access content from that user eye view.
```

Comment:
```suggestion
     * By impersonating a user SlingClient can access content from the perspective of that user.
```
Also, can you add a bit more info about what that means? Permissions, other session cookies (i assume not), etc?
And something like "any subsequent requests made from this <code>SlingClient</code> will... "